obspy/signal/invsim.py

```diff
@@ -763,6 +764,76 @@ def estimate_wood_anderson_amplitude_using_response(response, amplitude,
     return wa_ampl
 
 
+def _paz_to_freq_resp(freqs, zeros, poles, scale_fac):
+    b, a = scipy.signal.ltisys.zpk2tf(zeros, poles, scale_fac)
+    if not isinstance(a, np.ndarray) and a == 1.0:
+        a = [1.0]
+    return scipy.signal.freqs(b, a, freqs * 2 * np.pi)[1]
+
+
+def _unpack_paz(x, numpoles, numzeros):
+    poles_real = np.array(x[:numpoles * 2:2])
+    poles_imag = np.array(x[1:numpoles * 2:2])
+    poles = poles_real + 1j * poles_imag
+    zeros_real = np.array(x[numzeros * 2:numzeros * 2 + numzeros * 2:2])
+    zeros_imag = np.array(x[numzeros * 2 + 1:numzeros * 2 + numzeros * 2:2])
+    zeros = zeros_real + 1j * zeros_imag
+    scale_fac = x[-1]
+    return poles, zeros, scale_fac
+
+
+def _response_misfit(reference, other):
+    """
+    Calculate a misfit measure from two complex response vectors
+    """
+    # just the simple euclidean distance for now. tweaking this might improve
+    # the fitting maybe..
+    misfit = scipy.spatial.distance.euclidean(reference, other)
+    return misfit
+
+
+def optimize_paz(frequencies, response, numpoles, numzeros, num_tries=3,
+                 maxiter=100000, eps=1e-12):
+    frequencies = np.array(frequencies)
+    response = np.array(response)
+
+    def minimize(_var):
+        poles, zeros, scale_fac = _unpack_paz(
+            _var, numpoles=numpoles, numzeros=numzeros)
+        new_response = _paz_to_freq_resp(
+            freqs=frequencies, zeros=zeros, poles=poles, scale_fac=scale_fac)
+        misfit = _response_misfit(response, new_response)
+        return misfit
+
+    results = []
+    for i in range(num_tries):
+        out = scipy.optimize.minimize(
+            fun=minimize,
+            method="BFGS",
+            x0=np.random.random(numpoles * 2 + numzeros * 2 + 1),
```

Comment:
The starting point is fairly critical.  When I run this on an STS-2 if change the guess to 
`guess = [-0.037, -0.037, -0.037, +0.037, 0, 0, 0, 0]`

where `x0=guess`
![COR_good_guess](https://user-images.githubusercontent.com/2599501/83166740-944bf580-a0cc-11ea-9bce-ff3f1f2e858f.png)

However, if I leave in the initial guess as random, then I get the following:
`guess = np.random.random(numpoles * 2 + numzeros * 2 + 1)`
![COR_random](https://user-images.githubusercontent.com/2599501/83167024-f3aa0580-a0cc-11ea-9f3f-e5ce0ca25247.png)

This makes me think you want to allow for an initial guess.

Reply:
Yeah, initial guess is where a lot of smart brains could go into, right now its far from ideal with just random numbers